### ISSUE TITLE
Better way to move from Summary to Detail 

### ISSUE THREAD

**Issue by bph** [WordPress/gutenberg#50778]
### Description

When I added the block to the canvas, I could see the summary part and started typing, but when previewing everything I typed landed in the open. It doesn't behave like other blocks. It's pure text block that doesn't end. 

There doesn't seem to be a method to move to the 'hidden block' part without using the mouse. I would have expected that a `<Tab>` or `<Enter>` would get me to the next step.
Only on the second try I actually saw the 'Type "/" to add a hidden block'

After I entered my first paragraph, and hit enter, the 'hidden' part was missing from the next block prompt and I didn't know whether I was still adding to the Details section or already outside the Details/Summary block.

### Step-by-step reproduction instructions

- Enable the _Experiment > Details block_
- Open a post 
- Add a details block
- Add a summary
- Add two paragraphs to the Details section. 

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/39980/ef6c2b96-8451-4973-a83c-101cb3adf6cf

### Environment info

WordPress 6.2.1
Gutenberg 15.8
Chrome Version 113.0.5672.92
MacOS Venture 13.3.1 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by liviopv** [comment]
This behavior is still present in the stable release of the block in GB 16. While the paragraph nudge is visually helpful, I still expected to be able to hit Enter in the Summary to exit it and start typing in a nested block. Instead, it added a line break in the Summary section.

**Comment by himanshupathak95** [comment]
Thanks @bph for raising the issue. 

I tested it on the latest trunk and found the same issue. While I am perfectly in favour of using the `Enter` keypress to enter into the hidden block, I think the current configuration also makes sense. 

Here is a screencast of the same - 

https://github.com/user-attachments/assets/0886cbb7-7686-4932-b494-1e220ebe78bd

**Comment by westonruter** [comment]
I also would have expected the Enter key to (1) expand the Details block _once_, (2) move to an empty paragraph inside the Details block. 

It is confusing how every time I hit enter it opens and closes the Details. 

It's also confusing how every time I click on the Details block, it causes it to expand/collapse:

https://github.com/user-attachments/assets/7e56d5b8-f1c3-4d3c-b584-70a12402621d

I would expect the block to always be expanded when either the summary or hidden blocks are focused, unless the "Open by default" toggle is enabled, in which case the block would _always_ be expanded in the editor.
